### DISCUSSION TITLE
issue: setValue doesn't clear errors

### DISCUSSION THREAD

**Discussion by anshour** [react-hook-form/react-hook-form#7897] [Q&A] [answered]
### Version Number

7.22

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-datepicker-range-j1kryq?file=/src/App.js

### Steps to reproduce

1. Fill the form
2. Click on setError button (it will make startOrderDate error)
3. Make changes in order date input
4. The error doesn't disappear even we change the value

### Expected behaviour

The error disappear / cleared if we invoke setValue method

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
incorrect API usage, for clear custom error, you should use `clearError` https://react-hook-form.com/api/useform/clearerrors API.

**Reply by anshour** [reply]
Yes, but when i registered input with register function, i don't need to invoke clearErrors method. i think it is better to automatically trigger clearError when we call setValue

**Comment by anshour** [comment]
And i wanna ask is that ok to set the value input with watch function and bind onChange event with setValue?
```
<ReactDatePicker selected={watch('date')}  onChange={(date)=>{setValue(date)}}/>
```

**Reply by Moshyfawn** [reply]
It's not the intended use. In case of controlled components _(storing the input's state in React and passing it into the input)_, it recommended to use `Controller` / `useController` API instead

**Reply by nvmnghia** [reply]
what about `setValue(key, value, { shouldValidate: true })`?
